src/Core/src/Layouts/GridLayoutManager.cs

```diff
@@ -951,29 +973,79 @@ static void ExpandStars(double targetSize, double currentSize, Definition[] defs
 					}
 				}
 
-				foreach (var definition in defs)
+				// Use density-aware distribution for pixel-perfect proportional allocation when density != 1.0
+				// For density = 1.0, preserve original behavior to maintain fractional values
+				if (Math.Abs(density - 1.0) > 0.0001)
 				{
-					if (definition.IsStar)
+					var proportionalStarDefinitions = defs.Where(d => d.IsStar).ToArray();
+					var proportionalPortions = new double[proportionalStarDefinitions.Length];
+					
+					for (int i = 0; i < proportionalStarDefinitions.Length; i++)
 					{
-						// Skip the star rows/columns whose minimums are at or higher than the target sizes
+						var definition = proportionalStarDefinitions[i];
 						double fullTargetSize = targetStarSize * definition.GridLength.Value;
-
+						
 						if (definition.MinimumSize < fullTargetSize)
 						{
-							// Figure out how small this definition is relative to the total difference,
-							// and use that to determine how much of the available space this definition gets
-
-							// The goal is to have the definitions expand proportionate to their deficit from
-							// their full target sizes
-
 							var scale = (fullTargetSize - definition.MinimumSize) / totaldiff;
 							var portion = scale * availableSpace;
-							definition.Size = definition.MinimumSize + portion;
+							proportionalPortions[i] = definition.MinimumSize + portion;
+						}
+						else
+						{
+							proportionalPortions[i] = definition.MinimumSize;
+						}
+					}
+					
+					var proportionalTotalPixels = proportionalPortions.Sum() * density;
+					var proportionalPixelAllocations = DensityValue.DistributePixels(proportionalTotalPixels, density, proportionalPortions);
+					
+					for (int i = 0; i < proportionalStarDefinitions.Length; i++)
+					{
+						proportionalStarDefinitions[i].Size = DensityValue.FromPixels(proportionalPixelAllocations[i], density);
+					}
```

Comment:
I want this code to be the code that runs regardless of what the density value is. 

We shouldn't process the calculation of the sizes differently if the density equals 1 vs if it's greater than 1.